Project does not build when GOOS=darwin

/kind bug

**What happened?**

The project does not compile on MacOS / darwin because there is no code to implement the necessary `nodeService` methods like there is for [Linux](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/pkg/driver/node_linux.go) and [Windows](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/pkg/driver/node_windows.go).

**What you expected to happen?**

Code compiles well enough to run e2e tests.

**How to reproduce it (as minimally and precisely as possible)?**

```shell
$ GOOS=darwin go build -o bin/aws-ebs-csi-driver ./cmd/
# github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/driver
pkg/driver/mount.go:63:9: cannot use &NodeMounter{…} (value of type *NodeMounter) as Mounter value in return statement: *NodeMounter does not implement Mounter (missing method GetDeviceNameFromMount)
pkg/driver/node.go:190:19: d.findDevicePath undefined (type *nodeService has no field or method findDevicePath)
pkg/driver/node.go:340:21: d.IsBlockDevice undefined (type *nodeService has no field or method IsBlockDevice)
pkg/driver/node.go:346:19: d.getBlockSizeBytes undefined (type *nodeService has no field or method getBlockSizeBytes)
pkg/driver/node.go:360:23: d.findDevicePath undefined (type *nodeService has no field or method findDevicePath)
pkg/driver/node.go:375:17: d.getBlockSizeBytes undefined (type *nodeService has no field or method getBlockSizeBytes)
pkg/driver/node.go:480:20: d.IsBlockDevice undefined (type *nodeService has no field or method IsBlockDevice)
pkg/driver/node.go:486:23: d.getBlockSizeBytes undefined (type *nodeService has no field or method getBlockSizeBytes)
pkg/driver/node.go:592:19: d.findDevicePath undefined (type *nodeService has no field or method findDevicePath)
pkg/driver/node.go:697:14: d.preparePublishTarget undefined (type *nodeService has no field or method preparePublishTarget)
pkg/driver/node.go:697:14: too many errors
```

**Anything else we need to know?**:

I don't actually want to run the driver on MacOS, but I do want to run e2e tests to validate my installation of the driver.
Currently I can work around this by running the tests in a container.

**Environment**
- Kubernetes version (use `kubectl version`): EKS 1.22
- Driver version: v1.17.0

--- a/pkg/mounter/mount_unsupported.go
+++ b/pkg/mounter/mount_unsupported.go
@@ -1,0 +1,98 @@
+//go:build darwin
+// +build darwin
+
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mounter
+
+import (
+	"fmt"
+	mountutils "k8s.io/mount-utils"
+)
+
+const (
+	stubMessage = "nodeService is unsupported for this platform"
+)
+
+/*
+NOTE: This is stub implementation of nodeService so that maintainers without access to a Linux/Windows workstation can
+run driver e2e tests.
+*/
+
+func NewSafeMounter() (*mountutils.SafeFormatAndMount, error) {
+	return nil, fmt.Errorf("NewSafeMounter is not supported on this platform")
+}
+
+func NewSafeMounterV2() (*mountutils.SafeFormatAndMount, error) {
+	return nil, fmt.Errorf("NewSafeMounterV2 is not supported on this platform")
+}
+
+func (m *NodeMounter) FindDevicePath(devicePath, volumeID, partition, region string) (string, error) {
+	return stubMessage, fmt.Errorf(stubMessage)
+}
+
+func (m *NodeMounter) PreparePublishTarget(target string) error {
+	return fmt.Errorf(stubMessage)
+}
+
+func (m *NodeMounter) IsBlockDevice(fullPath string) (bool, error) {
+	return false, fmt.Errorf(stubMessage)
+}
+
+func (m *NodeMounter) GetBlockSizeBytes(devicePath string) (int64, error) {
+	return 1, fmt.Errorf(stubMessage)
+}
+
+func (m *NodeMounter) appendPartition(devicePath, partition string) string {
+	return stubMessage
+}
+
+func (m NodeMounter) GetDeviceNameFromMount(mountPath string) (string, int, error) {
+	return stubMessage, 0, fmt.Errorf(stubMessage)
+}
+
+func (m NodeMounter) IsCorruptedMnt(err error) bool {
+	return false
+}
+
+func (m *NodeMounter) MakeFile(path string) error {
+	return fmt.Errorf(stubMessage)
+}
+
+func (m *NodeMounter) MakeDir(path string) error {
+	return fmt.Errorf(stubMessage)
+}
+
+func (m *NodeMounter) PathExists(path string) (bool, error) {
+	return false, fmt.Errorf(stubMessage)
+}
+
+func (m *NodeMounter) Resize(devicePath, deviceMountPath string) (bool, error) {
+	return false, fmt.Errorf(stubMessage)
+}
+
+func (m *NodeMounter) NeedResize(devicePath string, deviceMountPath string) (bool, error) {
+	return false, fmt.Errorf(stubMessage)
+}
+
+func (m *NodeMounter) Unpublish(path string) error {
+	return fmt.Errorf(stubMessage)
+}
+
+func (m *NodeMounter) Unstage(path string) error {
+	return fmt.Errorf(stubMessage)
+}

